### PR TITLE
TS-73-Refactor-Schedule-page

### LINKED ISSUE #73: Refactor: Schedule page
###### _Acceptance Criteria:_
Fixed all bugs from the schedule table, created a separate card component for drag and drop.

Estimate:  _2 days_ | _3 days_

### PR DESCRIPTION
**Pull request by HiroFive**

### CODE REVIEW

File: frontend/src/App.scss [hunk 1]
```diff
@@ -162,21 +164,21 @@ body {
   }
 
   .info-btn {
-    color: $good-color;
+    color: $create-color;
```
Comment (ViktoriiaKosovan) [inline frontend/src/App.scss:167]: Could you name color as it looks like?

File: frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import i18n from 'i18next';
+import { CircularProgress } from '@material-ui/core';
+import { getClassScheduleListService } from '../../services/classService';
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js]: need to change to saga

File: frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                },
+            );
+        });
+    });
+
+    useEffect(() => {
+        setScheduleLoading(true);
+        getAllScheduleItems();
+        fetchEnabledGroupsStart();
+        showListOfRoomsService();
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js:72]: as above

File: frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoard.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        additionClassName,
+        currentSemester,
+        groupId,
+        scheduleItems,
+        openDialogWithData,
+        checkScheduleItemAvailability,
+    } = props;
+    const { classId, dayName, week } = lesson;
+    const [className, setClassName] = useState('');
+    const [boardFill, setBoardFill] = React.useState({});
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoard.js]: delete React from useState

File: frontend/src/styles/colors.scss [hunk 5]
```diff
@@ -1,29 +1,35 @@
...
 $gray: #6b7280;
 $light-gray: #d1d5db;
 $light-clear-gray: #edeef1;
 $snowy-white: #ffffff;
+$yellow: #f7f7c1;
 $blue: #1976d2;
 $blue-hover: #155da5;
+$light-blue: #4c94f6;
 $dark-blue: #3f51b5;
 $selected-blue: #0000001a;
```
Comment (ViktoriiaKosovan) [inline frontend/src/styles/colors.scss:12]: this color similar with number 2

File: frontend/src/styles/colors.scss [hunk 6]
```diff
@@ -1,29 +1,35 @@
...
-$light-forest: #028802;
-$forest: #184b44;
+$green: #028802;
+$dark-green: #184b44;
+
+$create-color-light: #90ee90;
+$create-color: $green;
+$edit-color: $blue;
+$delete-color: $red;
```
Comment (ViktoriiaKosovan) [inline frontend/src/styles/colors.scss:19]: why you use $red twice? use once

File: frontend/src/styles/colors.scss [hunk 7]
```diff
@@ -1,29 +1,35 @@
...
+$warning-color: $yellow;
```
Comment (ViktoriiaKosovan) [inline frontend/src/styles/colors.scss:20]: as above
this file need to refactor

File: frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js (hunk 3 above)
Comment (HiroFive) [reply frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js:72]: I will fix showListOfRoomsService() in next pull request

File: frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoard.js (hunk 4 above)
Comment (HiroFive) [reply frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoard.js]: removed

File: frontend/src/styles/colors.scss (hunk 6 above)
Comment (HiroFive) [reply frontend/src/styles/colors.scss:20]: removed

File: frontend/src/styles/colors.scss (hunk 5 above)
Comment (HiroFive) [reply frontend/src/styles/colors.scss:12]: removed

File: frontend/src/styles/colors.scss (hunk 7 above)
Comment (HiroFive) [reply frontend/src/styles/colors.scss:20]: I've created something like constants for certain events for better scalability in the future

File: frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+const ScheduleItem = (props) => {
+    const {
+        deleteScheduleItem,
+        checkRoomAvailability,
+        itemData,
+        getLessonsByGroupId,
+        selectByGroupId,
+        openDialogWithData,
+    } = props;
+    const [anchorEl, setAnchorEl] = React.useState(null);
```
Comment (Niknikson) [inline frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js]: remove React

File: frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js (hunk 2 above)
Comment (HiroFive) [reply frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js]: changed

**Review by Yashchemskyi-Oleksandr — APPROVED** [review]

File: frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        if (window.innerWidth < EDIT_SCHEDULE_MIN_WINDOW_SIZE) {
+            setIsHiddenSchedule(true);
+        } else {
+            setIsHiddenSchedule(false);
+        }
+    };
+
+    useEffect(() => {
+        window.addEventListener('resize', handleResize);
+        handleResize();
```
Comment (Niknikson) [inline frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js:43]: window.addEventListener('resize', handleResize)
    return () => {
      window.removeEventListener('resize', handleResize)
    }

File: frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+            </section>
+        );
+    }
+    return (
+        <>
+            <section className="schedule-control-panel">
+                <section className="card schedule-table ">
+                    {scheduleLoading ? (
+                        <CircularProgress className="loading-circle" />
+                    ) : (
```
Comment (Niknikson) [inline frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js:103]: if scheduleLoading true you can return  <CircularProgress className="loading-circle" />

File: frontend/src/helper/schedule.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+        const modifiedLesson = lesson;
+        const { hours } = modifiedLesson;
+
+        if (!isEmpty(items)) {
+            filteredLesson = items.filter((item) => item.lesson.id === lesson.id);
+        }
+
+        for (let i = 0; i < hours - filteredLesson.length; i += 1) {
+            lessonItem.push(modifiedLesson);
+        }
```
Comment (Niknikson) [inline frontend/src/helper/schedule.js:18]: for  => forEach ?

File: frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js (hunk 9 above)
Comment (HiroFive) [reply frontend/src/components/EditCurrentSchedule/EditCurrentSchedulePage.js:43]: fixed

**Review by Niknikson — APPROVED** [review]

File: frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js (hunk 8 above)
Comment (HiroFive) [reply frontend/src/components/EditCurrentSchedule/ScheduleBoard/ScheduleBoardItem.js]: done

**Review by ViktoriiaKosovan — APPROVED** [review]

**Pull request closed without merging**